http-clients/aws-crt-client/src/main/java/software/amazon/awssdk/http/crt/AwsCrtAsyncHttpClient.java

```diff
@@ -253,6 +256,9 @@ public CompletableFuture<Void> execute(AsyncExecuteRequest asyncRequest) {
         paramNotNull(asyncRequest.requestContentPublisher(), "RequestContentPublisher");
         paramNotNull(asyncRequest.responseHandler(), "ResponseHandler");
 
+        MetricCollector metricCollector = asyncRequest.metricCollector().orElseGet(NoOpMetricCollector::create);
```

Comment:
Should we not call report metrics if `asyncRequest.metricCollector()` is not present or it's an instance of NoOpMetricCollector (the default SDK metric collector) to avoid potential performance degradation? We skip metrics collection in netty and apache http client fwiw

https://github.com/aws/aws-sdk-java-v2/blob/master/http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/NettyRequestMetrics.java#L46-L49

Reply:
I was thinking noop metric collector was pretty cheap due to it being a no-op? What's the performance degradation?

Reply:
yes looking at NoOpMetricCollector.java, it's just a function call. It seems to me that less branching is a better trade-off than avoiding a java function call that the compiler probably optimizes out anyways. From a performance perspective, an instruction cache miss due to branching is likely to have worse performance profiles than an empty function call. Will defer to your judgement however,

Reply:
Yeah, @millems did benchmark before: https://github.com/aws/aws-sdk-java-v2/pull/1960#issue-664833683 Disabling is slightly better

Reply:
i'll happily remove for the sake of consistency with the rest of the code base

Reply:
plus it would save a JNI call, which is certainly non-trivial in this case